### PR TITLE
Migrate db hostname script

### PR DESCRIPTION
**Pull request by wjordan**
One-off script to automate migrating the live application to a new db hostname with minimal downtime.

### CODE REVIEW

File: bin/oneoff/migrate_db.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  # 3. Promote the new DB instance to master.
+  puts "Promoting new DB instance to master..."
+  run "echo 'CALL mysql.rds_reset_external_master();' | mysql -u'#{writer.user}' -p'#{writer.password}' -h '#{writer.host}'"
+
+  # 4. Start applications on all frontends. [live-site downtime ends]
+  fanout 'sudo chef-client'
+end
+
+main
```
Comment (ashercodeorg) [inline bin/oneoff/migrate_db.rb]: Is it worth using `cdo/only_one` in this script?

File: bin/oneoff/migrate_db.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  servers = JSON.parse(`../bin/ls_frontend_servers -j`).values
+  run "pdsh -w #{servers.join(',')} -R ssh #{cmd}"
+end
+
+def main
+  db_host=ARGV[0]
+  environment=ARGV[1]
+
+  # 1. Stop applications on all frontends. [live-site downtime begins]
+  fanout 'sudo service dashboard stop; sudo service pegasus stop'
```
Comment (ashercodeorg) [inline bin/oneoff/migrate_db.rb]: If this script is to live in our repository, I'd strongly encourage a 'Do you mean to do this?' confirmation before causing live-site downtime.

File: bin/oneoff/migrate_db.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+# Use the `pdsh` utility and `ls_frontend_servers` script to 'fan out' a command via ssh to all front-ends.
+def fanout(cmd)
+  puts "Running #{cmd}..."
+  servers = JSON.parse(`../bin/ls_frontend_servers -j`).values
+  run "pdsh -w #{servers.join(',')} -R ssh #{cmd}"
+end
+
+def main
+  db_host=ARGV[0]
+  environment=ARGV[1]
```
Comment (ashercodeorg) [inline bin/oneoff/migrate_db.rb]: Perhaps worth doing some input validation?

**Comment by jeremydstone** [comment]
LGTM

**Pull request merged**